src/main/java/uk/ac/cam/cl/dtg/isaac/dos/eventbookings/PgEventBookings.java

```diff
@@ -621,6 +621,48 @@ public Iterable<EventBooking> findAllByEventIdAndStatus(final String eventId, @N
     }
   }
 
+  /**
+   * Find all bookings for events.
+   *
+   * @param eventIds - the event of interest.
+   * @return an iterable with all the events matching the criteria.
+   * @throws SegueDatabaseException - if an error occurs.
+   */
+  @Override
+  public Iterable<EventBooking> findAllByEventIds(final List<String> eventIds) throws SegueDatabaseException {
+    StringBuilder sb = new StringBuilder();
+    sb.append("SELECT event_bookings.* FROM event_bookings JOIN users ON users.id=user_id WHERE event_id IN (");
```

Comment:
If you pass an empty list to this function, then the resulting SELECT statement will be `SELECT ... WHERE event_id IN () AND ...`. But would that work? I cannot remember if it is valid to have a condition that checks if a column can be found in an empty list.